pkg/tests/accounts_test.go

```diff
@@ -851,3 +864,233 @@ func Test_NegativeBIC_LockedAccountOutput(t *testing.T) {
+
+		ts.AssertTransactionsInCacheBooked([]*iotago.Transaction{tx1.Transaction}, true, node1)
+
+		latestParent = ts.CommitUntilSlot(block1Slot, block1)
+
+		wallet2BIC -= iotago.BlockIssuanceCredits(block1.WorkScore()) * iotago.BlockIssuanceCredits(ts.API.ProtocolParameters().CongestionControlParameters().MinReferenceManaCost)
+
+		// The outputID of wallet1 and wallet2 account should remain the same as neither was successfully spent.
+		ts.AssertAccountData(&accounts.AccountData{
+			ID:              wallet1.BlockIssuer.AccountID,
+			Credits:         accounts.NewBlockIssuanceCredits(wallet1BIC, 0),
+			OutputID:        wallet1AccountOutput.OutputID(),
+			ExpirySlot:      iotago.MaxSlotIndex,
+			BlockIssuerKeys: wallet1.BlockIssuer.BlockIssuerKeys(),
+		}, ts.Nodes()...)
+		ts.AssertAccountData(&accounts.AccountData{
+			ID:              wallet2.BlockIssuer.AccountID,
+			Credits:         accounts.NewBlockIssuanceCredits(wallet2BIC, block1Slot),
+			OutputID:        wallet2AccountOutput.OutputID(),
+			ExpirySlot:      iotago.MaxSlotIndex,
+			BlockIssuerKeys: wallet2.BlockIssuer.BlockIssuerKeys(),
+		}, ts.Nodes()...)
+	}
+
+	block2Slot := latestParent.ID().Slot()
+
+	//TRY TO SPEND THE BASIC OUTPUT FROM AN ACCOUNT ADDRESS
+	{
+		block2Commitment := node1.Protocol.MainEngineInstance().Storage.Settings().LatestCommitment().Commitment()
+
+		tx2 := wallet1.SendFundsFromAccount(
+			"TX2",
+			"Genesis:2",
+			lo.PanicOnErr(block2Commitment.ID()),
+			"TX1:0",
+		)
+
+		// Wallet 2, which has non-negative BIC issues the block.
+		ts.IssueBasicBlockAtSlotWithOptions("block2", block2Slot, wallet2, tx2, mock.WithStrongParents(latestParent.ID()), mock.WithSlotCommitment(block2Commitment))
+
+		ts.AssertTransactionsInCacheInvalid([]*iotago.Transaction{tx2.Transaction}, true, node1)
+
+		latestParent = ts.CommitUntilSlot(block2Slot, latestParent)
+
+		// The outputID of wallet1 and wallet2 account should remain the same as neither was successfully spent.
+		ts.AssertAccountData(&accounts.AccountData{
+			ID:              wallet1.BlockIssuer.AccountID,
+			Credits:         accounts.NewBlockIssuanceCredits(wallet1BIC, 0),
+			OutputID:        wallet1AccountOutput.OutputID(),
+			ExpirySlot:      iotago.MaxSlotIndex,
+			BlockIssuerKeys: wallet1.BlockIssuer.BlockIssuerKeys(),
+		}, ts.Nodes()...)
+		ts.AssertAccountData(&accounts.AccountData{
+			ID:              wallet2.BlockIssuer.AccountID,
+			Credits:         accounts.NewBlockIssuanceCredits(wallet2BIC, block1Slot),
+			OutputID:        wallet2AccountOutput.OutputID(),
+			ExpirySlot:      iotago.MaxSlotIndex,
+			BlockIssuerKeys: wallet2.BlockIssuer.BlockIssuerKeys(),
+		}, ts.Nodes()...)
+	}
+
+	block3Slot := latestParent.ID().Slot()
+
+	// UNLOCK THE ACCOUNT
+	// The locked wallet 2 is preparing and signs the transaction, but it's issued by wallet 1 whose account is not locked.
```

Comment:
The comment is reversed, should be wallet 1 creates tx and wallet 2 issues. Same with below comment.